R/matching.R

```diff
@@ -112,44 +132,30 @@ closest <- function(x, table, tolerance = Inf, ppm = 0,
         stop("'tolerance' hat to be of length 1 or equal to 'length(table)'")
```

Comment:
could we change this now to have length equal `length(x)`?

Reply:
Of course. As I said this is now easily possible but will be a backward-incompatible change and may already break something (for `Spectra`).

Reply:
I don't think that it will brake anything at present - the *default* use case is to have a `tolerance` of length 1 and a `ppm` of length 1 - supporting `tolerance` of length > 1 is a *bonus*. So I think it is safe to change, but of course we can also leave it as it is. I have no strong opinion on that.

Reply:
I changed `tolerance` in relation to `x` now (`length(x) == length(tolerance)`).